### PR TITLE
{bio}[gompi/2020b] IQ-TREE v2.1.2

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
@sassy-crick `Eigen/3.3.9` is only currently available in `GCCcore/11.2.0` and `10.3.0`. Please add the correct dependency.

**Comment by sassy-crick** [comment]
> @sassy-crick `Eigen/3.3.9` is only currently available in `GCCcore/11.2.0` and `10.3.0`. Please add the correct dependency.

I agree, However, the build fails with this version of Eigen with:
`error: eigen_assert_exception is not a member of Eigen`
Which got solved by the later version of Eigen. How shall we solve that?

**Comment by boegel** [comment]
@sassy-crick Since Eigen is just a build dependency, you can resolve the issue by adding an easyconfig for Eigen 3.3.9 with `GCCcore/10.2.0` (maybe best in a separate PR).

This shouldn't cause trouble, because Eigen is a header-only library, so it's a build-only dependency (won't result in having two Eigen versions loaded at runtime).

**Comment by sassy-crick** [comment]
@boegel Thanks for your advice, I have done that as requested. 
- [ ] Requires #14009

**Review by boegel — APPROVED** [review]
lgtm

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.6
See https://gist.github.com/ec4ffa88aec172db10090300fb865646 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/5642062eb06b241af40e3795ce83e01d for a full test report.

File: easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-2.1.2-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+easyblock = 'CMakeMake'
+
+name = 'IQ-TREE'
+version = '2.1.2'
+
+# HTTPS is not working
+homepage = 'http://www.iqtree.org/'
+description = """Efficient phylogenomic software by maximum likelihood"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
```
Comment (boegel) [inline easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-2.1.2-foss-2020b.eb:10]: @sassy-crick We can/should drop this to `gompi`, see https://github.com/sassy-crick/easybuild-easyconfigs/pull/6

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
node2635.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/2e469d6a8f893c2019f1aa020a2c50fb for a full test report.

**Comment by boegel** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**